Modules/_cursesmodule.c

```diff
@@ -3361,10 +3397,15 @@ _curses_initscr_impl(PyObject *module)
 
     SetDictInt("LINES", LINES);
     SetDictInt("COLS", COLS);
+#undef SetDictInt
 
-    winobj = (PyCursesWindowObject *)PyCursesWindow_New(win, NULL);
+    _cursesmodule_state *state = get_cursesmodule_state(module);
+    PyCursesWindowObject *winobj = (PyCursesWindowObject *)PyCursesWindow_New(state, win, NULL);
+    CHECK_NOT_NULL_OR_ABORT(winobj);
     screen_encoding = winobj->encoding;
     return (PyObject *)winobj;
```

Comment:
I suggest adding an empty line (after return, before the label) for readability.